include/logging/log_core.h

```diff
@@ -223,7 +223,7 @@ extern "C" {
 /******************************************************************************/
 /****************** Defiinitions used by minimal logging **********************/
 /******************************************************************************/
-void log_minimal_hexdump_print(int level, const char *data, size_t size);
+void log_minimal_hexdump_print(int level, const void *data, size_t size);
```

Comment:
have you only tried `LOG_MINIMAL` setting? It is strange that other functions did not complain about  implicit pointer conversion; `log_hexdump_from_user` , `log_hexdump_sync` , `log_hexdump` all those functions take `const u8_t *`. For consistency it would be good to have all those functions taking same argument.

Reply:
Ah, oops, I was grepping for `char* data` and missed the `u8_t*`s. Switched the other functions over to void* as well.

Reply:
can you explain why `const u8_t *` was not an issue before? If `const char *data` was causing an issue `const u8_t *` should too.

Reply:
This only affects people who are doing all of the following:

* using C++
* using LOG_HEXDUMP
* trying to hexdump a u8_t* with LOG_MINIMAL , or a char* with the other configs
* not casting the pointer or using -fpermissive

I'm guessing that the set of people who check all of those boxes and care enough to complain about it is just me, so far :-)

Reply:
ok, maybe c++ compiler treats things a bit different. `__LOG_HEXDUMP` macro is pure C code so all paths are compiled disregarding current logger configuration. Because of that compiler would rise a warning independent of LOG_MINIMAL setting.